syft/pkg/package.go

```diff
@@ -33,6 +33,14 @@ func (p *Package) OverrideID(id artifact.ID) {
 }
 
 func (p *Package) SetID() {
+	if p.Name == "" {
+		log.Warnf("%s: missing package name, that is necessary for further metadata extraction. Please take a look at %s", p.FoundBy, p.Locations.ToSlice())
```

Comment:
adding `p.Locations.ToSlice()` to the warning log to inform user where the problem is.